### DISCUSSION TITLE
Quarkus native and XML parsing gives

### DISCUSSION THREAD

**Discussion by nicklasweasel** [quarkusio/quarkus#43851] [Community]
I have a simple test case starting from the AWS quarkus-lambda archetype that does a simple DocumentBuilderFactory/XPathFactoryDocument extraction of a node value. When compiling to native and run, I get a 

> Caused by: java.lang.RuntimeException: exception creating new instance for pool
> at java.xml@21.0.4/com.sun.org.apache.xml.internal.utils.ObjectPool.getInstance(ObjectPool.java:145)
> at java.xml@21.0.4/com.sun.org.apache.xml.internal.utils.StringBufferPool.get(StringBufferPool.java:45)
> at java.xml@21.0.4/com.sun.org.apache.xml.internal.dtm.ref.dom2dtm.DOM2DTM.getStringValue(DOM2DTM.java:822)
> at java.xml@21.0.4/com.sun.org.apache.xpath.internal.objects.XNodeSet.getStringFromNode(XNodeSet.java:219)
> at java.xml@21.0.4/com.sun.org.apache.xpath.internal.objects.XNodeSet.str(XNodeSet.java:282)
> at java.xml@21.0.4/com.sun.org.apache.xpath.internal.jaxp.XPathImplUtil.getResultAsType(XPathImplUtil.java:151)
> at java.xml@21.0.4/com.sun.org.apache.xpath.internal.jaxp.XPathImpl.evaluate(XPathImpl.java:137)

my first two problems mentioned files so I got around them with

`quarkus.native.additional-build-args=-H:IncludeResourceBundles=com.sun.org.apache.xml.internal.res.XMLErrorResources\\,com.sun.org.apache.xerces.internal.impl.msg.XMLMessages`

but on this I'm at the end of good ideas

**Comment by gsmet** [comment]
I would expect GraalVM to take care of the internal JDK classes properly. Any chance you could put together a small Maven reproducer (ideally that we can run in a local environment)?

**Reply by nicklasweasel** [reply]
Thanks for the reply, got it working. I *thought* I had an XML implementation included through the quarkus-jaxb but it didn't stick so it was probably that.
